library/src/main/kotlin/com/vinted/preferx/EntityPrefSerializer.kt

```
@@ -24,11 +24,11 @@ class EntityPrefSerializer<T>(
         return if (cachedValue != null && cachedValue.string == string) {
             cachedValue.value ?: default
         } else {
-            val value = entitySerializer.fromString(string, clazz)
+            val value = serializer.fromString(string, clazz) as? T
             cached = SoftReference(Cache(value, string))
             value ?: default
         }
     }
 
-    private class Cache<out T>(val value: T, val string: String)
+    private class Cache<out T>(val value: T?, val string: String)
```

Why nullable type here? Maybe it is not required to even store null as a value into cache?

---

Good catch. I missed that. If you are serializing, and serializers give the wrong type of object, you should crash with the proper message instead of storing a null value. Silently storing null values doesn't help find a real problem.